docs/2_skeleton.md

```diff
@@ -0,0 +1,169 @@
+
+     Example: _The patch disables ads for an app. The app regularly updates and the code of the app mutates heavily. In that case the patch might not be compatible for future, untested versions of the app. To discourage the use of the app with other versions than the versions, this patch was confirmed to work on, it is constrained to those versions only._
+
+   - Patches can be annotated with `@Version`.
+
+     > Currently, this annotation does not serve any purpose, but is added to patches by convention, in case a use case has been found.
+
+2. 🏗️ Patch class
+
+   ```kt
+   class DisableAdsPatch : BytecodePatch( /* Parameters */ ) {
+     // ...
+   }
+   ```
+
+   Usually, patches consist out of a single class. The class can be used to create methods and fields for the patch, or provide a framework for other patches, in case it is meant to be used as a dependency patch.
+
+   The [ReVanced patches](https://github.com/revanced/revanced-patches) follow a convention to name the class of patches:
+
+   Example: _The class for a patch which disables ads should be called `DisableAdsPatch`, for a patch which adds a new download feature it should be called `DownloadsPatch`._
+
+   Each patch implicitly implements the [Patch](https://github.com/revanced/revanced-patcher/blob/main/src/main/kotlin/app/revanced/patcher/patch/Patch.kt#L15) interface when extending off [ResourcePatch](https://github.com/revanced/revanced-patcher/blob/main/src/main/kotlin/app/revanced/patcher/patch/Patch.kt#L35) or [BytecodePatch](https://github.com/revanced/revanced-patcher/blob/main/src/main/kotlin/app/revanced/patcher/patch/Patch.kt#L42). The current example extends off `BytecodePatch`:
+
+   ```kt
+   class DisableAdsPatch : BytecodePatch( /* Parameters */ ) {
+     // ...
+   }
+   ```
+
+   If the patch extends off `ResourcePatch`, it is able to **patch resources** such as `XML`, `PNG` or similar files. On the other hand, if the patche extends off `BytecodePatch`, it is able to **patch the bytecode** of an app. If a patch needs access to the resources and the bytecode at the same time. Either can use the other as a dependency. **Circular dependencies are unhandled.**
+
+3. 🏁 The `execute` method
+
+   ```kt
+   override fun execute(context: BytecodeContext): PatchResult {
+     // ...
+   }
+   ```
+
+   The `execute` method is declared in the `Patch` interface and therefore part of any patch:
+
+   ```kt
+   fun execute(context: /* Omitted */ T): PatchResult
+   ```
+
+   It is the **first** method executed when running the patch. The current example extends off `BytecodePatch`. Since patches which extend on it can interact with the bytecode, the signature for the execute method when implemented requires a [BytecodeContext](https://github.com/revanced/revanced-patcher/blob/main/src/main/kotlin/app/revanced/patcher/data/Context.kt#L23) as a parameter:
```

Comment:
Same thing here.
```suggestion
   It is the **first** method executed when running the patch. The current example extends off `BytecodePatch`. Since patches that extend on it can interact with the bytecode, the signature for the execute method when implemented requires a [BytecodeContext](https://github.com/revanced/revanced-patcher/blob/d2f91a8545567429d64a1bcad6ca1dab62ec95bf/src/main/kotlin/app/revanced/patcher/data/Context.kt#L23) as a parameter:
```